### ISSUE TITLE
Bindingsource not assigned

### ISSUE THREAD

**Issue by r-schaefer** [dotnet/winforms#8908]
### Environment

Visual Studio 2022 - 17.5.3

### .NET version

7.0

### Did this work in a previous version of Visual Studio and/or previous .NET release?

Yes

### Issue description

Assigning a class as data source to a binding source doesn't seems to work within the designer. The data source class can be selected but isn't assigned as data source.

see attached sample project and screenshots
![screen1](https://user-images.githubusercontent.com/17042351/228161370-51a561bb-d807-4708-a5e2-045f0ef36c91.png)
![screen2](https://user-images.githubusercontent.com/17042351/228161395-e83c11df-0e27-4de2-aae8-230c7849f1fc.png)
[WinFormsAppTestBindingSource.zip](https://github.com/dotnet/winforms/files/11086587/WinFormsAppTestBindingSource.zip)

### Steps to reproduce

- Create a new winform
- drop / drop a bindingsource to the form
- create a simple class with one property / member
- open datasource of the bindingsource  property and assign the class with "Add new object data source"
- select the simple class and press ok

- expected: the class is assigned as data source
- current behaviour: the class isn't assigned

**Comment by elachlan** [comment]
CC: @KlausLoeffelmann (designer + databinding)

**Comment by dreddy-work** [comment]
@Olina-Zhang, can you check if this worked in any versions of .NET core?

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang, can you check if this worked in any versions of .NET core?

It can reproduce in all previous .Net versions. Here is the simple repro steps:
 1. Create a Winforms .Net core application
 2. Add a bindingSource control to form designer, go to this control's properties window, and find DataSource property, open its dropdown, click "Add new Object Data Source..." link
 3. In "Add and update object data source" dialog, uncheck "Only show project defined types", check any one of DataSource in table, click OK button
 
Result: The chosen dataSource is not updated in DataSource property automatically, it needs to choose it manually. See following gif. @r-schaefer is it your problem? I verified your sample, can find the defined dataSource in DataSource dropdown, but need to choose manually. @dreddy-work @KlausLoeffelmann Could you please see if it is a real issue?
![source](https://user-images.githubusercontent.com/26474449/228407264-8286a27e-0232-4bba-8b92-724ec1ebeb55.gif)

**Comment by r-schaefer** [comment]
Yes, I can confirm that this is the problem I reported (Meanwhile I also discovered the solution to manually add the data source).
